Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -280,18 +283,21 @@ import BraintreeDataCollector
                        error.code == ASWebAuthenticationSessionError.canceledLogin.rawValue {
                         if self.returnedToAppAfterPermissionAlert == true {
                             // User tapped system cancel button in browser
-                            let eventName = "ios.\(paymentType.stringValue).authsession.browser.cancel"
-                            self.apiClient.sendAnalyticsEvent(eventName)
+                            self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.authsessionBrowserCancel)
                         } else {
                             // User tapped system cancel button on permission alert
-                            let eventName = "ios.\(paymentType.stringValue).authsession.alert.cancel"
-                            self.apiClient.sendAnalyticsEvent(eventName)
+                            self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.authsessionAlertCancel)
                         }
+                        self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserLoginCanceled)
+                        completion(nil, BTPayPalError.canceled)
+                        return
                     }
-                    
+                   
                     // User canceled by breaking out of the PayPal browser switch flow
                     // (e.g. System "Cancel" button on permission alert or browser during ASWebAuthenticationSession)
-                    completion(nil, BTPayPalError.canceled)
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationFailed)
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
+                    completion(nil, BTPayPalError.webSessionError)
```

Comment:
We should consider passing the error that we get back from `ASWebAuthenticationSession` into our error. It may make troubleshooting an error much easier if we know what `ASWebAuthenticationSession` returned to us. We should be able to do something like `BTPayPalError.webSessionError(error)`

Reply:
Have we confirmed that the comment above this is not true and this shouldn't be a cancel case? If so, we should remove these comments.

Reply:
I had originally made the new error with associated type, but I wanted to handle the error from authenticationSession?.start() failure the same way and that failure to start and that function only returns a boolean, 
not an error thrown from the websession.  I could make it another BTPayPalError but I thought it would be confusing to have another type of error like failedToLaunchBrowser. I think two types of cancel (from tapping on cancel buttons) are handled in the closure and tapping on the link to cancel does return an error and falls through and gets handled in handleBrowserSwitchReturn.

Reply:
We also shouldn't send a `browserPresentationFailed` event here. That only needs to happen based on the boolean below from `authenticationSession?.start()` (line 310).

Reply:
oh yes, I should fix the comments. They are errors other than cancel at this point.

Reply:
The error block from line 281 covers error cases other than cancel. I think there can be "webContentError", "configurationError", "preFlightError" or just "failed". So in the case it's an error other than cancel,  Jax and I thought it might be grouped with "BrowserPresentationFailed" but I get that would be confusing because it has presented. 
 
 I think the control of failure of start() goes back to the completion handler even if I don't do the return explicitly here.
So that fail would be the ."failed" in the completion handler.

Reply:
For us, in BTPayPalClient, we cannot be 100% sure that the webView was presented in the completion handler.